### ISSUE TITLE
[Feat. Req.] Search Functionality

### ISSUE THREAD

**Issue by guy-who-googles** [sezanzeb/key-mapper#207]
I remap a lot of keys. When I want to tweak my config, it takes a while to find what I'm looking for. Is search functionality of interest for this project?

P.S. This tool is truly awesome!

**Comment by sezanzeb** [comment]
Search function for macros and keys?

Other than the autocompletion of the gui and the docs there is nothing

**Comment by guy-who-googles** [comment]
Yes, it would be useful for users with many entries. For example, if I had the following entries:
| Key            | Macro            |
| ---            | ---              |
| apostrophe+e   | hold(mouse(up,2))|
| apostrophe+d   | hold(mouse(down,2))|
| apostrophe+s   | hold(mouse(left,2))|
| apostrophe+f   | hold(mouse(right,2))|

then searching (through a search bar or shortcut) for "down" would only show the `apostrophe+d` entry

**Comment by sezanzeb** [comment]
ahh, I see. You want to filter your mapping table.

Or maybe something similar to a file manager:

![image](https://user-images.githubusercontent.com/28510156/141658977-30febad4-6bd9-425c-b713-a50c92fb1c70.png)

with the search term on the bottom left, which then jumps to the first match. Using arrow keys then cycles through the matches

**Comment by guy-who-googles** [comment]
Assuming users don't have too many entries, jumping to matches seems reasonable and is probably easier to implement. If they have tons of entries, it may be less friendly to cycle through many jumps.

Filtering would probably make more sense for many entries but would have the addiional implementation cost of changing the output of mapping table.
